### ISSUE TITLE
routing not working fully (upgraded from 7 to 9)

### ISSUE THREAD

**Issue by maapteh** [zeit/next.js#8666]
# Bug report
In my current App, routing is not working anymore for one Link out of three in production only (development everything works). I use Apollo hooks for data retrieval and rendering components SSR or CSR by passing SSR to their options. 

When i request the particular page directly i see this is working. But when i shift from one page to this it just hangs and not throwing anything. It seems that in new NextJS something has been made to do get into this state?. In version 7 i was totally in control and the same setup worked.

In development mode everything works, in production build not and with side effects.

## To Reproduce
1. Go to https://graphql-schiphol.herokuapp.com (is on Heroku, needs some time to spin also for api part) its a static page
2. Then try to go to products (will not work, its a dynamic page fed by Apollo hooks for data just like home). Going to Info (no dynamic data) works
3. Open te same link in new tab and all of a sudden you can go to every page and stuff is being loaded correctly.

## Expected behavior
To be able to go to every page. It seems the logic for a component to be rendered SSR or CSR is somehow interfering with the new NextJS pages setup? I can't find any reason only that it's not working. The pages are defined overhere https://github.com/maapteh/graphql-modules-app/tree/master/packages/app/pages

Hopefully its a bug and i made mistake from the docs and not the new setup in the framework :( 

The example is showing i have to wrap each page withApollo while in the 7 setup i wrapped the apollo Provider on app level (so only once, see [here](https://github.com/maapteh/graphql-modules-app/blob/0.0.1/packages/graphql-app/pages/_app.js)) i dont think its a good example in your examples now by doing it per page, but maybe its needed for nextjs :(

**Comment by maapteh** [comment]
And after the weekend all started to work again. Don't know if i upgraded 3rd libs, so closing now.

**Comment by maapteh** [comment]
@timneutkens for you there is still a question in the last paragraph (Nextjs HOC per page level, why not on app level once, my whole app is Apollo nothing static about it).

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
